### PR TITLE
fix: Get calico namespace automatically

### PR DESCRIPTION
**Pull request by AbelHu**
Get calico namespace automatically

### CODE REVIEW

**Comment by AbelHu** [comment]
cc @song-jiang

File: parts/windows/windowscalicofunc.ps1 [hunk 1]
```diff
@@ -62,11 +62,26 @@ function Start-InstallCalico {
...
     .\install-calico.ps1
     popd
+}
+
+function GetCalicoNamespace() {
+    param(
+      [parameter(Mandatory=$false)] $KubeConfigPath = "c:\\k\\config"
+    )
+
+    $name=c:\k\kubectl.exe --kubeconfig=$KubeConfigPath get ns calico-system
```
Comment (song-jiang) [inline parts/windows/windowscalicofunc.ps1:80]: Just wondering with AKS windows nodes, does `c:\k\config` and `c:\k\kubectl.exe` always exist?

File: pkg/templates/templates_generated.go [hunk 2]
```diff
@@ -5603,13 +5603,28 @@ function Start-InstallCalico {
     SetConfigParameters -RootDir $CalicoDir -OldString "CALICO_NETWORKING_BACKEND=`+"`"+`"vxlan`+"`"+`"" -NewString "CALICO_NETWORKING_BACKEND=`+"`"+`"none`+"`"+`""
     SetConfigParameters -RootDir $CalicoDir -OldString "KUBE_NETWORK = `+"`"+`"Calico.*`+"`"+`"" -NewString "KUBE_NETWORK = `+"`"+`"azure.*`+"`"+`""
 
-    GetCalicoKubeConfig -RootDir $CalicoDir -CalicoNamespace "kube-system" -SecretName "calico-windows"
+    $calicoNs = GetCalicoNamespace
+    GetCalicoKubeConfig -RootDir $CalicoDir -CalicoNamespace $calicoNs -SecretName "calico-windows"
```
Comment (song-jiang) [inline pkg/templates/templates_generated.go:5607]: Just a reminder that once Calico version on Linux side  is updated to the latest, we would not need to apply a work around manifest to create `calico-windows` secret. The secret name should be updated to `calico-node` which is the default.

File: parts/windows/windowscalicofunc.ps1 (hunk 1 above)
Comment (AbelHu) [reply parts/windows/windowscalicofunc.ps1:80]: Yes for now.

File: pkg/templates/templates_generated.go (hunk 2 above)
Comment (AbelHu) [reply pkg/templates/templates_generated.go:5607]: Yes. I got your confirmation. Is anything needed to be updated here?

Comment (song-jiang) [reply pkg/templates/templates_generated.go:5607]: No. Looking good!

File: parts/windows/windowscalicofunc.ps1 (hunk 1 above)
Comment (mainred) [reply parts/windows/windowscalicofunc.ps1:80]: Did you hit such an issue?
`Error from server (Forbidden): namespaces "calico-system" is forbidden: User "nodeclient" cannot get resource "namespaces" in API group "" in the namespace "calico-system"`

File: pkg/templates/templates_generated.go (hunk 2 above)
Comment (mainred) [reply pkg/templates/templates_generated.go:5607]: I know you have somewhere to track this future change, but could we add a comment here?

File: parts/windows/windowscalicofunc.ps1 (hunk 1 above)
Comment (mainred) [reply parts/windows/windowscalicofunc.ps1:80]: What will happen if the command returns an error?

File: pkg/templates/templates_generated.go (hunk 2 above)
Comment (AbelHu) [reply pkg/templates/templates_generated.go:5607]: In this guidance, we need step 4 as a workaround to test calico windows with old Linux calico version. We do not need it for using calico windows v3.17.1 with calico Linux v3.17.1. I think that we do not need such a comment for a step only used by an old test.

File: parts/windows/windowscalicofunc.ps1 (hunk 1 above)
Comment (AbelHu) [reply parts/windows/windowscalicofunc.ps1:80]: @mainred, if your cluster does not have the namespace `calico-system`, you will hit this issue. `name` will be empty and "kube-system" will be returned.
In Windows provisioning scripts, any thrown error will be caught by `catch` in `kuberneteswindowssetup.ps1`. There is no difference here.

Comment (AbelHu) [reply parts/windows/windowscalicofunc.ps1:80]: I have some misunderstanding for your error message. @mainred yes. it is forbidden so we cannot get namespace via kubectl.exe inside Windows nodes.

**Comment by AbelHu** [comment]
We cannot get namespace via kubectl

**Comment by AbelHu** [comment]
Updated and reuse this PR.

**Pull request closed without merging**